addons/dialogue_nodes/objects/Character.gd

```diff
@@ -6,3 +6,7 @@ extends Resource
 @export var name : String = ''
 @export var image : Texture2D
 @export var color : Color = Color.WHITE
+
+var translated_name : String :
+	get:
+		return tr(name)
```

Comment:
I could move that `tr()` to the DialogueParser but then we would still need the `translated_name` property (or we overwrite the `name` property but that just feels wrong) since the `dialogue_processed` signal emitted by the DialogueParser contains the Character object and not just the simple translated name. And it would have to be overwritten in every `_process_dialogue` (since theoretically the language could be changed which is why i can't just check if the property is set). 

I could also instead of modifying the Character object create a new copy of it in each call to `_process_dialogue` and only translate the copys name. (and then either overwrite the `name` or set the `translated_name` if we keep the latter.)
 
Alternatively i can call `tr()` when speaker_label.text is set to speaker.name in the DialogueBox / DialogueBubble.

Which do you prefer?

Reply:
The last option seems great. We can leave the translation call to be handled by the DialogueBox/DialogueBubble nodes. Maybe even put the translation code for the dialogue and option text to DialogueBox/DialogueBubble nodes?

Reply:
Ok got it, just one more question. Should the `_on_dialogue_processed` inside of the DialogueBox and DialogueBubble emit `dialogue_processed` with the untranslated or translated data or maybe even both?